File: src/main/java/com/anod/appwatcher/accounts/AccountManager.java
package com.anod.appwatcher.accounts;

import android.accounts.Account;
import android.content.Context;

import com.anod.appwatcher.Preferences;

/**
 * @author alex
 * @date 2015-03-03
 */
public class AccountManager {

    private final android.accounts.AccountManager mAccountManager;
    private final Preferences mPreferences;
    private Account[] mAccounts;
    private Account mCurrentAccount;

    public AccountManager(Context context) {
        mAccountManager = android.accounts.AccountManager.get(context);
        mPreferences = new Preferences(context);
        mAccounts = mAccountManager.getAccountsByType(AuthTokenProvider.ACCOUNT_TYPE);
        mCurrentAccount = mPreferences.getAccount();
    }

    public Account[] getAccounts() {
        return mAccounts;
    }

    public Account getCurrentAccount() {
        return mCurrentAccount;
    }

    public boolean hasAccounts() {
        return mAccounts.length > 0;
    }

    public void reload()
    {
        mAccounts = mAccountManager.getAccountsByType(AuthTokenProvider.ACCOUNT_TYPE);
        mCurrentAccount = mPreferences.getAccount();
    }

    public Account getAccount(int idx) {
        return mAccounts[idx];
    }

    public void saveCurrentAccount(Account acc) {
        mPreferences.updateAccount(acc);
        mCurrentAccount=acc;
    }
}

File: src/main/java/com/anod/appwatcher/accounts/AccountManager.kt
package com.anod.appwatcher.accounts

import android.accounts.Account
import android.content.Context

import com.anod.appwatcher.Preferences

/**
 * @author alex
 * *
 * @date 2015-03-03
 */
class AccountManager(context: Context, private val prefs: Preferences) {

    private val mAccountManager: android.accounts.AccountManager = android.accounts.AccountManager.get(context)
    var accounts: Array<Account> = mAccountManager.getAccountsByType(AuthTokenProvider.ACCOUNT_TYPE)
        private set
    var currentAccount: Account? = prefs.account
        private set

    fun hasAccounts(): Boolean {
        return accounts.isNotEmpty()
    }

    fun reload() {
        accounts = mAccountManager.getAccountsByType(AuthTokenProvider.ACCOUNT_TYPE)
        currentAccount = prefs.account
    }

    fun getAccount(idx: Int): Account {
        return accounts[idx]
    }

    fun saveCurrentAccount(acc: Account) {
        prefs.account = acc
        currentAccount = acc
    }
}
